### PR TITLE
feat: Reorganize password onboarding

### PR DESCRIPTION
**Pull request by ptbrowne**
- Use a Stepper to show onboarding progress
- Check if vault has been configured to bypass first steps
- Check hint to show a message if already configured
- Show warning in stepper if vault configured but hint is not configured

### CODE REVIEW

**Review by Crash-- — COMMENTED** [review]
Can you fix the tests and ask again for review?

File: src/components/HintStep.jsx [hunk 1]
```diff
@@ -13,14 +13,13 @@ import Spinner from 'cozy-ui/transpiled/react/Spinner'
 import VerticallyCentered from './VerticallyCentered'
 import BarTitle from 'BarTitle'
 
-const DumbHintPage = props => {
-  const { client, history } = props
+const DumbHintStep = props => {
+  const { client } = props
```
Comment (Crash--) [inline src/components/HintStep.jsx]:
```suggestion
  const { client, goToNextStep } = props
```

File: src/components/HintStep.jsx [hunk 2]
```diff
@@ -13,14 +13,13 @@ import Spinner from 'cozy-ui/transpiled/react/Spinner'
...
   const { t } = useI18n()
   const [hint, setHint] = useState('')
   const [saving, setSaving] = useState(false)
-  const [loading, setLoading] = useState(true)
 
-  const goToNextStep = () => history.push('/installation')
+  const goToNextStep = () => props.goToNextStep()
```
Comment (Crash--) [inline src/components/HintStep.jsx]:
```suggestion
```

File: src/components/InstallationPage.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  useEffect(() => {
+    const fetch = async () => {
+      const hint = await fetchHint(client)
+      setHasHint(Boolean(hint))
+    }
+    fetch()
+  }, [activeStep])
+
+  return (
+    <div style={{ maxWidth: 800, margin: '0 auto', width: '100%' }}>
```
Comment (Crash--) [inline src/components/InstallationPage.jsx]: inline css ? 🤔

File: src/hint.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const fetchHint = async client => {
+  if (flag('passwords.force-no-hint')) {
+    return false
+  }
+  try {
+    const hint = await client
+      .getStackClient()
+      .collection('io.cozy.settings')
+      .get('hint')
```
Comment (Crash--) [inline src/hint.js:11]: can't you : 
```
client.query(client.get('io.cozy.settings', 'hint'))
``` 
?

File: src/components/App.jsx [hunk 5]
```diff
@@ -93,24 +40,32 @@ export const DumbApp = () => {
...
-              </Content>
-            </Main>
-            <IconSprite />
-            <Alerter />
-            <FlagSwitcher />
-          </Layout>
-        </HashRouter>
-      </MuiCozyTheme>
-    </BreakpointsProvider>
+    <BitwardenSettingsContext.Provider value={queryResult.data}>
```
Comment (Crash--) [inline src/components/App.jsx:50]: I like this idea 👍

**Review by Crash-- — APPROVED** [review]

File: src/hint.js (hunk 4 above)
Comment (ptbrowne) [reply src/hint.js:11]: No this is not possible as the hint does not return any data (204 No content), and it does not work with `query` in this case. Thank for asking though, I had not checked.

Comment (ptbrowne) [reply src/hint.js:11]: https://github.com/cozy/cozy-passwords/pull/25/commits/229f29e3604aa4e2d50396fc2176d63cc69eb78e makes it clearer.

**Pull request merged**